File: 0746_min_cost_climbing_stairs.swift
// https://leetcode.com/problems/min-cost-climbing-stairs/

class Solution {
    func minCostClimbingStairs(_ cost: [Int]) -> Int {
        var stack = [cost[0], cost[1]]

        if cost.count > 2 {
            for i in 2..<cost.count {
                let newCost = min(stack[0], stack[1]) + cost[i]
                stack[0] = stack[1]
                stack[1] = newCost
            }
        }

        return min(stack[0], stack[1])
    }
}

File: 0746_min_cost_climbing_stairs.kt
// https://leetcode.com/problems/min-cost-climbing-stairs/

import kotlin.math.min

class Solution {
    fun minCostClimbingStairs(cost: IntArray): Int {
        val n = cost.size
        if (n == 2) {
            return min(cost[0], cost[1])
        }
        
        for (i in 2 until n) {
            val currentCost = cost[i]
            cost[i] = currentCost + min(cost[i - 1], cost[i - 2])
        }
        return min(cost[n - 1], cost[n - 2])
    }
}
